ArkhamOverlay/Data/CardGroup.cs

```diff
@@ -0,0 +1,244 @@
+        public CardZone CardZone { get { return GetCardZone(0); } }
+
+        private bool _showCardZoneButtons;
+        /// <summary>
+        /// Whether card zones should be displayed on the UI
+        /// </summary>
+        public bool ShowCardZoneButtons { 
+            get => _showCardZoneButtons;
+            set {
+                _showCardZoneButtons = value;
+                NotifyPropertyChanged(nameof(ShowCardZoneButtons));
+            }
+        }
+
+        public bool Loading { get; internal set; }
+        public IEnumerable<CardTemplate> CardPool { get => from button in CardButtons.OfType<CardTemplateButton>() select button.CardTemplate; }
+
+        /// <summary>
+        /// Use the button context to find the button
+        /// </summary>
+        /// <param name="context">Information to find the button</param>
+        /// <returns>The button identified by the context- will be default if not found</returns>
+        internal IButton GetButton(IButtonContext context) {
+            if (context.CardGroupId != Id) {
+                return default(Button);
+            }
+
+            if (context.ButtonMode == ButtonMode.Pool) {
+                return (context.Index < CardButtons.Count) ? CardButtons[context.Index] : default(Button);
+            }
+
+            if (CardZone == default(CardZone)) {
+                return default(Button);
+            }
+
+            return context.Index < CardZone.Buttons.Count ? CardZone.Buttons[context.Index] : default(Button);
+        }
+
+
+        /// <summary>
+        /// Remove all CardTemplates from the pool and all card zones
+        /// </summary>
+        internal void ClearCards() {
+            CardButtons.Clear();
+            foreach (var cardZone in _cardZones) {
+                cardZone.Buttons.Clear();
+            }
+            NotifyPropertyChanged(nameof(CardButtons));
+        }
+
+        internal void LoadCards(IEnumerable<CardTemplate> cards) {
+            var clearButton = new ClearButton();
+
+            var playerButtons = new List<IButton> { clearButton };
+
+            if (_cardZones.Count > 0) {
+                _showCardZoneButton = new ShowCardZoneButton();
+                playerButtons.Add(_showCardZoneButton);
+                UpdateShowCardZoneButtonName();
+            }
+
+            playerButtons.AddRange(from card in SortCards(cards) select new CardTemplateButton(card));
+            CardButtons = playerButtons;
+            NotifyPropertyChanged(nameof(CardButtons));
+        }
+
+        private IEnumerable<CardTemplate> SortCards(IEnumerable<CardTemplate> cards) {
+            var firstCard = cards.FirstOrDefault();
+            if (firstCard == null) {
+                return cards;
+            }
+
+            //don't sort scenario cards- easier to find when acts/agendas are in order
+            if (firstCard.Type == CardType.Scenario) {
```

Comment:
Minor: this will break if scenario cards get into player's hands. Consider another method of applying this logic.

Reply:
This is some old logic, but I agree- added a todo.